enterprise_scale/construction_sets/aks/online/aks_secure_baseline/iac-pipeline.md

```diff
@@ -0,0 +1,118 @@
+|    1  | Foundation | Resource groups, Managed Identities, KeyVaults|
+|    2  | Shared Services | Log analytics and diagnostics|
+|    2  | Networking | Networking infrastructure including Vnets, subnets, firewalls, Application Gateways, etc.
+|    3  | AKS | Aks cluster | 
+|    4  | Flux | Flux V2 with GitSource and Kustomization pointing to the [infrastructure configurations](./cluster-baseline-settings) | 
+
+
+The pipeline requires the following secrets to be configured in the repository:
+| Secret | Description |Sample|
+|--------|-------------|------|
+|ENVIRONMENT| Any name of your environment|sandpit|
+|RESOURCE_PREFIX| Prefix for all names of the resources created by the pipeline|secureaks
+|SERVICE_PRINCIPAL| Service Principal which will be used to provision resources||
+|SERVICE_PRINCIPAL_PWD| Service Principal secret||
+|SUBSCRIPTION_ID| Azure subscription id||
+|TENANT| Azure tenant id||
+|FLUX_TOKEN| GitHub Token for Flux V2|| 
+
+
+To start the IaC pipeline execution, add to a PR or to an Issue on your repository "/deploy-all" comment. This comment will start deployment of all stages/levels in the pipeline from 0 (launchpad) to 4 (Flux). 
+In order to deploy specific levels add one or a few of the following comments: "/deploy-launchpad", "/deploy-foundation", "/deploy-networking", "/deploy-shared-services", "/deploy-aks", "/deploy-flux".
+
+In addition to the [GitHub Actions workflow](../../../../../.github/workflows/deploy-secure-aks-baseline.yaml), there is also an IaC [Azure Pipeline](../../../../../.pipelines/deploy-secure-aks-baseline.yaml) available to run on Azure DevOps orchestrator.
+
+![iac-azdo-pipeline](pictures/iac-azdo-pipeline.png)
+
+This pipeline can be started manually from Azure DevOps UI with specifying what stages/levels should be deployed. The pipeline expects the following environment variables to be configured in *iac-secure-caf* variable group:
+
+| Variable | Description |Sample|
+|--------|-------------|------|
+|ENVIRONMENT| Any name of your environment|sandpit|
+|PREFIX| Prefix for all names of the resources created by the pipeline|secureaks
+|ARM_CLIENT_ID| Service Principal which will be used to provision resources||
+|ARM_CLIENT_SECRET| Service Principal secret||
+|ARM_SUBSCRIPTION_ID| Azure subscription id||
+|ARM_TENANT_ID| Azure tenant id||
+|AZURE_SERVICE_NAME| ARM Service connection name|iac-caf-connection|
+|ROVER_IMAGE| Name and version of Rover Docker image|aztfmod/rover:0.14.8-2103.1601|
+|TF_VAR_github_owner| Owner of GitHub repo with cluster configurations |Azure|
+|TF_VAR_github_token| PAT with write access to the repo with cluster configurations  ||  
+
+## Deploying levels manually
+Alternatively you can deploy the construction set level by level manually with *rover*.
+
+```bash
+# Go to the AKS construction set folder 
+cd caf-terraform-landingzones-starter
+
+# Start Rover container
+docker run -it --rm -v $(pwd):/tf/caf --user 0  aztfmod/rover:0.14.8-2103.1601 bash
```

Comment:
If opened in container in vscode then rover is available in bash. 
The env variables below will be supplied by the rover too after rover login or az login. 
I will help to refine this guide